keyvalue/clients/src/main/java/tachyon/client/keyvalue/hadoop/KeyValueOutputCommitter.java

```diff
@@ -34,86 +41,64 @@
  * Extension of {@link FileOutputCommitter} where creating, completing, or deleting a
  * {@link KeyValueStores} in different phases of a job's or task's lifecycle is considered.
  * <p>
- * This committer is forced to be used in {@link KeyValueOutputFormat} (no matter what users have
- * set as the {@link org.apache.hadoop.mapred.OutputCommitter} in configuration) to merge the
- * key-value stores created by each Reducer into one key-value store under the MapReduce output
- * directory.
+ * This committer must be used along with {@link KeyValueOutputFormat} to merge the key-value stores
+ * created by each Reducer into one key-value store under the MapReduce output directory.
  */
+@PublicApi
 @ThreadSafe
-class KeyValueOutputCommitter extends FileOutputCommitter {
+public final class KeyValueOutputCommitter extends FileOutputCommitter {
   private static final KeyValueStores KEY_VALUE_STORES = KeyValueStores.Factory.create();
 
-  private TachyonURI getOutputURI(JobConf conf) {
-    return new TachyonURI(FileOutputFormat.getOutputPath(conf).toString());
-  }
-
-  private TachyonURI getTaskAttemptOutputURI(TaskAttemptContext context) throws IOException {
-    return new TachyonURI(getTaskAttemptPath(context).toString());
-  }
-
-  /**
-   * {@inheritDoc}
-   * <p>
-   * Calls {@link FileOutputCommitter#setupJob(JobContext)} first, and then creates an empty
-   * key-value store under the job's output directory.
-   */
-  @Override
-  public void setupJob(JobContext context) throws IOException {
-    super.setupJob(context);
-    try {
-      KEY_VALUE_STORES.create(getOutputURI(context.getJobConf())).close();
-    } catch (TachyonException e) {
-      throw new IOException(e);
-    }
-  }
-
-  /**
-   * {@inheritDoc}
-   * <p>
-   * Deletes the key-value store created in {@link #setupJob(JobContext)} first, and then
-   * calls {@link FileOutputCommitter#abortJob(JobContext, int)}.
-   */
-  @Override
-  public void abortJob(JobContext context, int runState) throws IOException {
-    // The output directory should exist since the store writer is just closed.
-    try {
-      KEY_VALUE_STORES.delete(getOutputURI(context.getJobConf()));
-    } catch (TachyonException e) {
-      throw new IOException(e);
+  private List<TachyonURI> getTaskTemporaryStores(JobConf conf) throws IOException {
+    TachyonURI taskOutputURI = KeyValueOutputFormat.getTaskOutputURI(conf);
+    Path outputPath = FileOutputFormat.getOutputPath(conf);
```

Comment:
the two paths are different, but your suggestion applies because we only need scheme information to let hdfs choose Tachyon as its FileSystem.